decidim-proposals/app/events/decidim/proposals/publish_proposal_event.rb

```diff
@@ -9,6 +9,32 @@ def resource_text
         resource.body
       end
 
+      def perform_translation?
```

Comment:
These methods (`perform_translation?, content_in_same_language?, and translation_missing`) seem copypasted. Can we move this to a concern? Something like `MachineTranslatedEvent` or whatever naming that you prefer.